### PR TITLE
fix fonts and DPI problem

### PR DESCRIPTION
**Pull request by Zioba**

### CODE REVIEW

File: sources/plugins/scg/scgwindow.cpp [hunk 1]
```diff
@@ -456,6 +456,42 @@ void SCgWindow::onExportImage()
     }
 }
 
+void SCgWindow::exportToPng(int index)
+{
+
+    SCgExportImage exportImage;
```
Comment (ValikVityaz) [inline sources/plugins/scg/scgwindow.cpp]:
```suggestion
{
    SCgExportImage exportImage;
```

File: sources/plugins/scg/scgwindow.cpp [hunk 2]
```diff
@@ -456,6 +456,42 @@ void SCgWindow::onExportImage()
...
+
+    SCgExportImage exportImage;
+    QString selectedFilter;
+
+    QString imageName = currentFileName();
+    QString fileName = QCoreApplication::applicationDirPath() + "/massExport/";
+    QDir().mkdir(fileName);
+    if (imageName.isEmpty())
+    {
+        fileName.append(QString::QString::number(index));
```
Comment (ValikVityaz) [inline sources/plugins/scg/scgwindow.cpp]:
```suggestion
        fileName.append(QString::number(index));
```

File: sources/plugins/scg/scgwindow.cpp [hunk 3]
```diff
@@ -456,6 +456,42 @@ void SCgWindow::onExportImage()
...
+    }
+    else
+    {
+        QStringList sl = imageName.split("/");
+        fileName.append(sl[sl.size()-1]);
```
Comment (ValikVityaz) [inline sources/plugins/scg/scgwindow.cpp]:
```suggestion
        fileName.append(sl.last());
```

File: sources/plugins/scg/scgwindow.cpp [hunk 4]
```diff
@@ -456,6 +456,42 @@ void SCgWindow::onExportImage()
...
+
+        if (info.suffix().isEmpty())
+        {
+            fileName += ".png";
+        }
+        else
+        {
+            // replace suffix if it not in selected filter
+            if (info.suffix() != "png")
+                fileName = fileName.left(fileName.size() - info.suffix().size()) + "png";
```
Comment (ValikVityaz) [inline sources/plugins/scg/scgwindow.cpp]:
```suggestion
                filename = info.dir().dirName() + "/" + info.completeBaseName() + ".png";
```
This is probably more clear.

File: sources/plugins/scg/scgwindow.cpp [hunk 5]
```diff
@@ -456,6 +456,42 @@ void SCgWindow::onExportImage()
...
+    if (imageName.isEmpty())
+    {
+        fileName.append(QString::QString::number(index));
+    }
+    else
+    {
+        QStringList sl = imageName.split("/");
+        fileName.append(sl[sl.size()-1]);
+    }
+    if (fileName.length() > 0)
```
Comment (ValikVityaz) [inline sources/plugins/scg/scgwindow.cpp]:
```suggestion
    if (!fileName.isEmpty())
```
For more clarity & to conform with lines 468, 481.

**Review by ValikVityaz — COMMENTED** [review]
Some simplifications

**Comment by deniskoronchik** [comment]
Fix comment in style like it was before

File: sources/kbe/interfaces/editorinterface.h [hunk 6]
```diff
@@ -104,6 +104,10 @@ class EditorInterface
      */
     virtual void activate(QMainWindow *window) { Q_UNUSED(window); mIsActivated = true; }
 
+    /*! Window export to png
+     */
+    virtual void exportToPng(int index) = 0;
```
Comment (deniskoronchik) [inline sources/kbe/interfaces/editorinterface.h]: Why we need that in an editor interface? For example why it should be implemented for any text editor?

File: sources/kbe/main.cpp [hunk 7]
```diff
@@ -25,6 +25,7 @@
 int main(int argc, char *argv[])
 {
     QApplication a(argc, argv);
+    QApplication::setAttribute(Qt::AA_Use96Dpi);
```
Comment (deniskoronchik) [inline sources/kbe/main.cpp:30]: Add comment with link, that describes this solution for HDPI?

File: sources/plugins/scg/scgwindow.cpp [hunk 8]
```diff
@@ -456,6 +456,42 @@ void SCgWindow::onExportImage()
...
 
+void SCgWindow::exportToPng(int index)
+{
+
+    SCgExportImage exportImage;
+    QString selectedFilter;
+
+    QString imageName = currentFileName();
+    QString fileName = QCoreApplication::applicationDirPath() + "/massExport/";
+    QDir().mkdir(fileName);
```
Comment (deniskoronchik) [inline sources/plugins/scg/scgwindow.cpp]:
```suggestion
    const QString kSubfolder("massExport");
    QDir dir(QCoreApplication::applicationDirPath());
    dir.mkpath(kSubfolder);   // also need to check if returns true. Some OS will have no permissions to create directory
    dir.cd(kSubfolder);
    
    QString fileName = dir.filePath(QString::number(index));
```

**Review by deniskoronchik — CHANGES_REQUESTED** [review]
Some improvements

File: sources/kbe/interfaces/editorinterface.h (hunk 6 above)
Comment (Zioba) [reply sources/kbe/interfaces/editorinterface.h]: We have no any text editors anymore.
As I know, there are no any plans to add any others editors in the future.

**Comment by deniskoronchik** [comment]
Let do not use apply proposals button on github. It generate commit for each change. 
Also please fixup commits `Update sources/plugins/scg/scgwindow.cpp` into one. And change commit message to `Review fixes`

**Review by deniskoronchik — CHANGES_REQUESTED** [review]
Don't use concat operations for path. Let use already implemented cross-platform classes like `QDir`

**Comment by Zioba** [comment]
Removed mass export feature from this PR
Renamed PR

**Review by ValikVityaz — APPROVED** [review]

**Comment by Zioba** [comment]
https://github.com/ostis-dev/kbe/pull/430#discussion_r536896560
https://www.charlesodale.com/setting-qt-to-ignore-windows-dpi-text-size-personalization/

File: sources/kbe/main.cpp (hunk 7 above)
Comment (Zioba) [reply sources/kbe/main.cpp:30]: https://www.charlesodale.com/setting-qt-to-ignore-windows-dpi-text-size-personalization/

Comment (deniskoronchik) [reply sources/kbe/main.cpp:30]: I mean add comment into code

**Review by deniskoronchik — CHANGES_REQUESTED** [review]
Small changes

**Review by KovalM — APPROVED** [review]

File: sources/kbe/main.cpp (hunk 7 above)
Comment (Zioba) [reply sources/kbe/main.cpp:30]: done

File: sources/kbe/main.cpp [hunk 9]
```diff
@@ -25,6 +25,8 @@
 int main(int argc, char *argv[])
 {
     QApplication a(argc, argv);
+    //prevention of text scaling https://www.charlesodale.com/setting-qt-to-ignore-windows-dpi-text-size-personalization/
```
Comment (deniskoronchik) [inline sources/kbe/main.cpp]:
```suggestion
    // Disable text scaling https://www.charlesodale.com/setting-qt-to-ignore-windows-dpi-text-size-personalization/

```

File: sources/kbe/main.cpp [hunk 10]
```diff
@@ -26,6 +26,9 @@ int main(int argc, char *argv[])
 {
     QApplication a(argc, argv);
 
+    //Disable text scaling https://www.charlesodale.com/setting-qt-to-ignore-windows-dpi-text-size-personalization/
```
Comment (deniskoronchik) [inline sources/kbe/main.cpp]:
```suggestion
    // Disable text scaling https://www.charlesodale.com/setting-qt-to-ignore-windows-dpi-text-size-personalization/

```

**Review by deniskoronchik — CHANGES_REQUESTED** [review]
Simple fix

**Review by deniskoronchik — APPROVED** [review]

**Pull request merged**